net/relayd/src/opnsense/mvc/app/controllers/OPNsense/Relayd/Api/SettingsController.php

```diff
@@ -0,0 +1,294 @@
+<?php
+
+/**
+ *    Copyright (C) 2018 EURO-LOG AG
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace OPNsense\Relayd\Api;
+
+use \OPNsense\Base\ApiMutableModelControllerBase;
+use \OPNsense\Core\Config;
+use \OPNsense\Relayd\Relayd;
+use \OPNsense\Base\UIModelGrid;
+
+/**
+ * Class SettingsController
+ * @package OPNsense\Relayd
+ */
+class SettingsController extends ApiMutableModelControllerBase
+{
+
+    static protected $internalModelName = 'relayd';
+    static protected $internalModelClass = '\OPNsense\Relayd\Relayd';
+
+    /**
+     * list with valid model node types
+     */
+    private $nodeTypes = array('general', 'host', 'tablecheck', 'table', 'protocol', 'virtualserver');
+
+    /**
+     * query relayd settings
+     * @param $nodeType
+     * @param $uuid
+     * @return result array
+     */
+    public function getAction($nodeType = null, $uuid = null)
```

Comment:
This can be folded as well since https://github.com/opnsense/core/pull/2159

@fabianfrz shows how to do it in https://github.com/opnsense/plugins/pull/544

Reply:
Hmm, getBase only works with array items.
Doesn't matter because I have to distinguish between 'general' as single item an the other array items anyway.

Besides it throws:
> Fatal error: Uncaught Error: Call to undefined method OPNsense\Base\FieldTypes\ContainerField::add() in /usr/local/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php:256

Seems that `$node = $mdl->add();` must be `$node = $mdl->Add();`